File: Backend/application_service/core/domain/contestant_application.go
package domain

import "github.com/google/uuid"

type ContestantApplication struct {
	ID                     uuid.UUID
	TeamNumber             int
	CompetitionID          uuid.UUID
	Competition            Competition
	ContestantID           uuid.UUID
	Contestant             Contestant
	AgeCategoryID          uuid.UUID
	AgeCategory            AgeCategory
	ApparatusAnnouncements []ApparatusAnnouncement
}

File: Frontend/Public/gco-public/src/app/model/core/contestant-application.ts
import { AgeCategory } from './age-category';
import { ApparatusAnnouncement } from './apparatus-announcement';
import {Competition} from './competition'
import {Contestant} from './contestant'

export interface ContestantApplication{
    id? : string;
    teamNumber : number;
    competition? : Competition;
    contestant : Contestant;
    ageCategory : AgeCategory;
    apparatusAnnouncements : ApparatusAnnouncement[];
}
  
export interface ContestantApplicationTable {
    displayedColumns: string[];
    dataSource: ContestantApplication[];
    selectedRow: ContestantApplication | null;
}

export interface ContestantApplicationPreview{
    id? : string;
    contestantFullName : string;
    teamNumber : number;
    ageCategory : AgeCategory;
    apparatusAnnouncements : ApparatusAnnouncement[];
  }

export interface ContestantApplicationPreviewTable {
    displayedColumns: string[];
    dataSource: ContestantApplicationPreview[];
    selectedRow: ContestantApplicationPreview | null;
}